app/models/components/course/duplicator_component_host.rb

```diff
@@ -0,0 +1,15 @@
+class Duplicator::ModelComponentHost
```

Comment:
I mean, having one host would make sense to me, since that deals with the course.

I'm not sure how much a model must implement for duplication for it to be duplicable. Even so, the duplication logic can be implemented as a concern and mixed into the model itself to be used by the model host, so I'm not sure why we need a separate model?

Unless you can elaborate on what you're thinking then we can iron out the problems.